src/cpp/include/openvino/genai/generation_config.hpp

```diff
@@ -126,88 +167,323 @@ class OPENVINO_GENAI_EXPORTS StructuredOutputConfig {
         std::string to_string() const {
             return "EBNF(\"" + value + "\")";
         }
+        std::string to_json() const {
+            return std::string("{\"type\": \"grammar\", \"grammar\": ") + format_for_json(value) + "}";
+        }
         bool operator==(const EBNF& other) const {
             return value == other.value;
         }
     };
 
+    struct ConstString {
+        std::string value;
+
+        ConstString() = default;
+        ConstString(const std::string& str) : value(str) {}
+        std::string to_string() const {
+            return "ConstString(\"" + value + "\")";
+        }
+        std::string to_json() const {
+            return std::string("{\"type\": \"const_string\", \"value\": ") + format_for_json(value) + "}";
+        }
+        bool operator==(const ConstString& other) const {
+            return value == other.value;
+        }
+    };
+
+    struct AnyText {
+        AnyText() = default;
+        std::string to_string() const {
+            return "AnyText()";
+        }
+        std::string to_json() const {
+            return "{\"type\": \"any_text\"}";
+        }
+        bool operator==(const AnyText& other) const {
+            return true;
+        }
+    };
+
+    struct QwenXMLParametersFormat {
```

Comment:
Seem to be a niche thing. Can you add a description a a link to learn more to the docstrings